### DISCUSSION TITLE
Adding buttons on the App via code

### DISCUSSION THREAD

**Discussion by frickyto** [terjeio/ioSender#389] [General]
Hello, I trying add some functions to the app, firstly I need, for example add a button to go home each axis, I add the button but I can´t find how to execute a gcode command, for example when I press a button.
I try check how to send a command when I write in MDI textbox but I can´t understand. Have any description about the code that I can read to understand it?.
I wan´t add buttons to the app and no add as Macro. I´d like that buttons stay close to the axis data and no separated
 Here I want to add a button to go home the axis
![image](https://github.com/terjeio/ioSender/assets/108746966/bd5b6ac8-8971-48c0-8f26-eefa1c03de91)

I try understand what happend with this part of code to use this for execute commands
![image](https://github.com/terjeio/ioSender/assets/108746966/c03b7a84-c4c6-4325-9d51-3b780a2eea29)

Regards!!!

**Comment by terjeio** [comment]
In the button eventhandler you can send a command like this:

`(DataContext as GrblViewModel).ExecuteCommand("my command");`

Tip: I use the [tag attribute](https://github.com/terjeio/ioSender/blob/614115d80c1f6c792112371e94f73741558cc407/CNC%20Controls/CNC%20Controls/DROControl.xaml.cs#L104C1-L105C67) to identify the axis, this way only one event handler is required.

**Reply by frickyto** [reply]
Thanks a lot Terjeio!!!!!!
It´s really fantastic app.
Regards!!!
